pypeit/metadata.py

```diff
@@ -204,35 +209,42 @@ def _build(self, files, strict=True, usrdata=None):
             filenames = np.asarray(data['filename'])
             bad_files = filenames[mjd == None]
             # Print status message
-            msgs_string = 'Validation of the time in your data failed for {:d} files'.format(len(bad_files)) + msgs.newline()
-            msgs_string += 'Continuing, but the following frames may be empty or have corrupt headers:' + msgs.newline()
+            msg = 'Time invalid for {0} files.\n'.format(len(bad_files))
+            msg += 'Continuing, but the following frames may be empty or have corrupt headers:\n'
             for file in bad_files:
-                msgs_string += '{0:s}'.format(file) + msgs.newline()
-            msgs.warn(msgs_string)
+                msg += '    {0}\n'.format(file)
+            msgs.warn(msg)
 
         # Return
         return data
 
+    # TODO: This method feels out of place in this class, but I don't
+    # really know how it's used.
+    # TODO: If you can only ever use the first element in the frames
```

Comment:
There was a time when manual extraction was going to come in via the `data` block.  That is gone.  And so too can this method be removed.

Reply:
Gone!